Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -686,7 +686,20 @@ protected class ClassWriter : IClassWriter {
 
     protected override void EmitNewArray(Type elmtType, IToken tok, List<Expression> dimensions, bool mustInitialize,
       ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
-      throw new NotImplementedException();
+      var initValue = mustInitialize ? DefaultValue(elmtType, wr, tok, true) : "None";
+      if (dimensions.Count == 1) {
+        // handle the common case of 1-dimensional arrays separately
+        wr.Write($"[{initValue} for _ in range");
+        TrParenExpr(dimensions[0], wr, false, wStmts);
+        wr.Write("]");
+      } else {
+        wr.Write("_dafny.newArray({0}", initValue);
```

Comment:
```suggestion
        wr.Write($"{DafnyRuntimeModule}.newArray({initValue}");
```
Ideally you would see those things yourself.